### PR TITLE
Added new checkbox to multiplayer game configure dialog to allow only registered users to join the game

### PR DESCRIPTION
**Pull request by pubudug**
Implemented "Make it possible to host a game which only registered players are allowed to join" from https://wiki.wesnoth.org/Easycoding

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
You didn't need to squash all your work into a single commit. In fact, I personally would have preferred if you hadn't, though it's not really a big deal.

**Comment by pubudug** [comment]
@CelticMinstrel Changed the commit message.

File: src/game_initialization/multiplayer_configure.cpp [hunk 1]
```diff
@@ -264,7 +269,10 @@ configure::~configure()
 	// don't set observers preference if disabled (for singleplayer)
 	if (observers_game_.enabled())
 		preferences::set_allow_observers(engine_.allow_observers());
-
+	// don't set registerd_users_only preference if disabled (for singleplayer)
+	if (registerd_users_only_.enabled())
```
Comment (CelticMinstrel) [inline src/game_initialization/multiplayer_configure.cpp]: It's minor, sure, but I dislike having misspelled variable names (the proper spelling is "registered").

**Comment by CelticMinstrel** [comment]
Well, I don't see any issues from glancing through the code, so that's good.

**Comment by pubudug** [comment]
@CelticMinstrel Fixed the spelling mistake

**Comment by CelticMinstrel** [comment]
Also, for the record - now that I've looked over the code, I think squashing it wasn't a problem. Each of the original commits was pretty tiny in the first place, so even the squashed whole isn't particularly large.

**Comment by gfgtdf** [comment]
things like speclling mistake corrections _should_ be squashed, if possible no swasich is only useful if those commits contains useful information (so like i you for exampel changed code becasue the previous behaviour you impleemted had bugs, then its useful to know that it was explicitly changes to the new code)

**Comment by pubudug** [comment]
@CelticMinstrel @gfgtdf Sqashed the spelling mistake commit.

**Comment by gfgtdf** [comment]
Hmm yes looks good to me, but there are still some issues to solve before merging:
1) Should this block only unregisterd user from joaind as player or also for observing? If the intention is for example to prevent letting 'levers' join your game then it makes sense to still allow unregisterd observers, but if the intention is like prevent spammers to join your game then it might be better to disaalowunregisterd observers.

2) The mp server currently has some bugs, it might be better to delay merging this pr until those are fixed to make debugging easier.

**Comment by pubudug** [comment]
1) I was thinking of both leavers and spammers, because the exact use case was not mentioned in the EasyCoding page. To handle the two separately, there would have to be two checkboxes and registering is not that hard a process if you want to join/observe games.

2) Is up to you I guess, but the fix isn't that large to cause any problems with debugging IMO.

**Comment by gfgtdf** [comment]
> 2) Is up to you I guess, but the fix isn't that large to cause any problems with debugging IMO.

hmm yes even though i don't think this will casue bugs or similar isuues its just easier if you are 100% sure that there are no other changes that could interfere with the debugged problem.

**Comment by gfgtdf** [comment]
mostlikeley the mentioned mp bugs were fixed in
https://github.com/wesnoth/wesnoth/commit/dce503ebec2c9d9c1ce574729b9321538ea8bb78
and
https://github.com/wesnoth/wesnoth/commit/722d186fed2bc4a34732b7865186d558fda92745

**Comment by Vultraz** [comment]
Any objections to merging?

**Comment by Vultraz** [comment]
Could you fix the merge conflicts?

**Comment by pubudug** [comment]
@Vultraz Fixed the conflict(was in the changelog).

**Comment by CelticMinstrel** [comment]
Yeah, changelog is a common source of merge conflicts.

**Comment by pubudug** [comment]
Maybe the feature request should be removed from https://wiki.wesnoth.org/Easycoding because someone else might also look into it not knowing it was done?

**Pull request merged**